kayak_core/src/fragment.rs

```diff
@@ -39,23 +39,15 @@ impl Widget for Fragment {
         // Do nothing.
     }
 
-    fn render(&mut self, context: &mut KayakContext) {
+    fn render(&mut self, context: &mut KayakContextRef) {
         let parent_id = self.get_id();
-        let tree = crate::WidgetTree::new();
-        tree.add(parent_id, None);
-
         if let Some(children) = self.children.take() {
-            children(tree.clone(), Some(parent_id), context);
+            let mut context = KayakContextRef::new(&mut context.context, Some(parent_id));
+            children(Some(parent_id), &mut context);
```

Comment:
Would we want to consider adding another method to `KayakContextRef` to do this for a user? It might be a common enough pattern.

Reply:
Or is this logic exclusive to `Fragment`?

Reply:
Because fragments are a bit weird the logic here is that we pass in a new context ref into the closure. I'm not sure we need to provide a different API. Fragments currently are the only way to resolve children I think..

Reply:
Makes sense, thanks!